### PR TITLE
Split the Mandrel and S2i builds to avoid disk space issues

### PR DESCRIPTION
**Pull request by cescoffier**

### CODE REVIEW

File: quarkus-mandrel-builder-image/mandrel-22.x.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+images:
+
+  - graalvm-version: 22.3.2.1-Final
+    java-version: 17
+    variants:
+      - sha: 7c9f0cfc18137fd1ae976d9a4aacbadf764a6a07c7691481a8d8d9b78903212a
+        arch: amd64
+      - sha: 95995f1b99eb9323e214ea147624651366129b9cfb3e07590eeac59f02f47ed4
+        arch: arm64
```
Comment (zakkak) [inline quarkus-mandrel-builder-image/mandrel-22.x.yaml:9]: There is no need to keep this around. 22.3.3.1 should be used instead.

File: quarkus-mandrel-builder-image/mandrel.yaml [hunk 2]
```diff
@@ -1,58 +1,5 @@
...
-  - graalvm-version: 23.0.1.2-Final
-    java-version: 20
-    tags: 23.0-java20, 23.0-jdk-20, jdk-20, jdk-20.0.2
-    variants:
-      - sha: be992959df83beb9f5fe0ffe4606d76b2deba6cb86322b9d3614f09a8a2fbbc8
-        arch: amd64
-      - sha: e6ed1f76292a5f44bff076225ebbb503b940180b14d6468d8722769ff7eb9e9e
-        arch: arm64
-
   - graalvm-version: 23.1.0.0-Final
```
Comment (zakkak) [inline quarkus-mandrel-builder-image/mandrel.yaml:3]: This one appears both here and in mandrel-23.0.x.yaml

I believe we don't need `mandrel-23.0.x.yaml` yet (?)

File: quarkus-mandrel-builder-image/mandrel.yaml [hunk 3]
```diff
@@ -1,58 +1,5 @@
...
-        arch: arm64
-
-  - graalvm-version: 23.0.1.2-Final
-    java-version: 17
-    tags: 23.0-java17, 23.0-jdk-17, jdk-17, jdk-17.0.8
-    variants:
-      - sha: cc73adfae094d05367e008d8ee3eebf7a08a3b45b504e55771dc739a7faabe3b
-        arch: amd64
-      - sha: 373d8341cfa678e3bdffeb8936c3edd8c353b19fdd54ba4cbe6711c6c1faa127
-        arch: arm64
```
Comment (zakkak) [inline quarkus-mandrel-builder-image/mandrel.yaml:45]: Please keep these around, 23.0 is an LTS release.

**Review by zakkak — CHANGES_REQUESTED** [review]
Thank you @cescoffier, I didn't find the time to do the cleanup myself.

Please see my comments about the Mandrel images.

File: quarkus-mandrel-builder-image/mandrel.yaml (hunk 2 above)
Comment (cescoffier) [reply quarkus-mandrel-builder-image/mandrel.yaml:3]: My mistake! 
The content of the file was not correct.

File: quarkus-mandrel-builder-image/mandrel.yaml (hunk 3 above)
Comment (cescoffier) [reply quarkus-mandrel-builder-image/mandrel.yaml:45]: Yes, my mistake.

**Review by zakkak — APPROVED** [review]
LGTM. We can also drop the `jdk-20` image

**Pull request merged**